File: 01_Array_and_String/Python Code/08_334. Increasing Triplet Subsequence.py
# Problem Link -> https://leetcode.com/problems/increasing-triplet-subsequence/description/

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        num1 = float('inf')
        num2 = float('inf')

        for num3 in nums:
            if num3 <= num1:
                num1=num3
            elif num3 <= num2:
                num2 = num3
            else: return True
        
        return False



# Time Complexity -> O(n)
# Space Complexity -> O(1)

File: 01_Array_and_String/C++ Code/08_334. Increasing Triplet Subsequence.cpp
// Problem Link -> https://leetcode.com/problems/increasing-triplet-subsequence/description/

class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int num1 = INT_MAX, num2 = INT_MAX;

        for(int i = 0; i < nums.size(); i++){
            int num3 = nums[i];

            if(num3 <= num1) num1 = num3;
            else if(num3 <= num2) num2 = num3;
            else return true;
        }

        return false;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(1)
